ddl/ddl_worker.go

```diff
@@ -621,12 +621,56 @@ func chooseLeaseTime(t, max time.Duration) time.Duration {
 	return t
 }
 
+// countForErrorOrPanic records the error count for ddl job.
```

Comment:
I think we need to share the logic, if we are trying to merge two functions. The new function is like:

```
	if err == nil {
       w.countForErrorOrPanic1(err, job)
    } else {
       w.countForErrorOrPanic2(err, job)
    }
```

Then why we did not call `countForErrorOrPanic1` and `countForErrorOrPanic2` directly?

IMO, a better abstraction is only extracting the logic of error count check, e.g.:

```
// true if the limit is exceeded, false otherwise
// possibly passing a closure or string for custom logging
func (w *worker) errorCountCheck(job) bool {
...
}
```

Reply:
make sense, I've split it into two func.